lib/ipmi_backend_utils.pm

```diff
@@ -528,4 +528,22 @@ sub set_grub_terminal_and_timeout {
     }
 }
 
+sub reconnect_when_ssh_console_broken {
+
+    # Switch to sol console to check serial console output
+    # It is useful in the case of host crash and reboot
+    record_info("WARN", "ssh connection is broken and switch to SOL console", result => 'fail');
+    select_console 'sol', await_console => 0;
+    save_screenshot;
+    wait_still_screen 5;
+    # Wait host bootup if it crashes
+    check_screen("linux-login", 120);
```

Comment:
It is obviously that ```check_screen("linux-login", 120)``` will delay test run significantly.

Reply:
Normally, there is no output to sol console after host bootup in our test.  The screen stops in the 'linux-login'. So `check_screen("linux-login", 120)` tooks little time. 
Only when the host crashed and rebooted, like the case in verification run #3, it need wait for the host to bootup completely. In this case,  `check_screen("linux-login", 120)` is to check if the host is up. It is necessary and the waiting time is worthy. without my ssh reconnection, you even don't know the host crashed.

Reply:
I do not think you need 
```
wait_still_screen 5;
check_screen("linux-login", 120);
```
You can use 
```
check_port_state
```
in lib/virt_autotest/utils.pm
The advantage of using this is there will no waste of time on wait for something uncertainty. ```check_port_state``` gives you the certainty. More clear logic and code.